contrib/dunst_xr_theme_changer.sh

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env sh
+###############################################################################
+#
+# This script creates a dunst user configuration file in
+# $XDG_CONFIG_HOME/dunst folder, changing dunst basic themming options
+# acording to your current X resources color palete.
+#
+# It gets your current user configuration file
+# -$XDG_CONFIG_HOME/dunst/dunstrc, which is copied from the default
+# config filein /usr/share/dunst/dunstrc if it does not exists- and
+# changes the defined attributes -declared furder on in the script as
+# $theme_attr_dict dictionary- to the new ones. Then it dumps the new
+# configuration to $user_themed_conf file.
+#
+###############################################################################
+
+script_name="$(basename $0)"
+
+# Function to get resource values
+xrdb_get () {
+    output=$(xrdb -q | grep -e "$1" | cut -f2)
+    default="$2"
+    printf '%s' "${output:-$default}"
+}
+
+# Default config dir and file
+def_conf_dir=/usr/share/dunst
+def_conf=$def_conf_dir/dunstrc
+
+# User config dir and file
+user_conf_dir=$XDG_CONFIG_HOME/dunst
+user_conf=$user_conf_dir/dunstrc
+
+# User themed config file
+user_themed_conf=$user_conf_dir/dunstrc_xr_themed
+
+# Check if the user config directory exists
+[ -d $user_conf_dir ] || {
+    printf '%s\n' "$user_conf_dir folder doesn't exist, creating..."
+    mkdir -p  $user_conf_dir
+}
+
+# If it exists, check if the user config file exists
+[ -f $user_conf ] || {
+    printf '%s does not exist.\n.' "$user_conf"
+    printf 'Copying default config to %s...\n' "$user_conf_dir"
+    cp $def_conf $user_conf_dir
```

Comment:
Should ensure that `$def_conf` exists in the first place, not every distro has the default dunstrc in `/usr/share/dunst/dunstrc`, in Debian for example it's in `/usr/share/doc/dunstrc.gz`

Reply:
Is there any other possible default directories to include in the search? Or should I just ask the user to change `$user_conf` path?

Reply:
I did some research and realized about the following:

|Distribution|Directory|file|
|:---:|:---:|:---:|
|Archlinux|`/usr/share/dunst`|`dunstrc`|
|Debian < 1.3.0-2 / Ubuntu < 18.04|`/usr/share/doc/dunst/`|`dunstrc.example.gz`|
|Debian >= 1.3.0-2 / Ubuntu >= 18.04|`/usr/share/doc/dunst/`|`dunstrc.gz`|